calculator.js

```diff
@@ -1,3 +1,23 @@
+/*
+<Eric>:
+Part 1:
+Ok, this is kind of advanced, but once you get it, you will be unstoppable.
+This is a 2 part-er.
+
+You're going to make a nice constant called `MATH_FUNCS`.
+It's going to be an object literal.
+You're going to name each property/key in this object literal to a name of each
+of your math functions below.
+Each property's value will be it's corresponding one-line fat-arrow function.
+
+```
+{
+  "add": (num1, num2) => return num1 + num2;,
+  "subtract" .... blah blah blah
```

Comment:
I'm following the syntax [in this MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer), which differs slightly from yours w/using quotations for property names. When using the "return num1 + num2" or whatever for each math function I'm getting an error with unreachable code in the next line, since it's following a return. 

Should I be following a different syntax? I[ noticed getter in the documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get), to reference an outside function in the object. Would this work?

Reply:
HMM.. it does sound like there's a syntax error. 

So I went and checked out what I wanted you to do in Quokka, (which is a free plugin for VSCode that allows you to run tiny javascripts in a little scratch area. It's a great testing tool to experiment with if you want to add it to your VSCode. )
Anyway, quokka is constantly evaluating your code as you right it. My last line, I wanted to see what the value of "adding" was. And sure enough, it was the right number! 

<img width="1381" alt="Screen Shot 2021-07-28 at 2 01 35 PM" src="https://user-images.githubusercontent.com/901315/127381057-6267bfc3-cae2-4842-a3b2-b09fa1579286.png">

Check out the example. It SHOULD work. :) 

Also, you're right; you _can_ use quotes for the properties, but it's nicer if you don't have to. :)

Let me know if the screenshot makes sense. This IS advanced! But this is a topic ("functions as first-class citizens") that will get you a LOT of leverage in JS once you get it. 

SIDENOTE: `getter` is typically used in classes. That's an advanced topic we'll talk about at a later date.

Reply:
And also, you might be looking at this and saying, WTF??!

But let me walk you through what line 20 is doing. 

```javascript
const adding = MATH_FUNCS["add"](3, 4)
```
Ok the first part is simple:
`const adding = `
You're just assigning the return value from anything right of the `=` to the variable `adding`. 

The second part:
`MATH_FUNCS["add"]`
Is where you're getting the value of the object-literal `MATH_FUNCS` for the property/key "add". 

This will return the value of "add" in MATH_FUNCS. 
This value is everything after the `:`. 
Which just so happens to be an "anonymous function". (i.e. it's a function, just hanging out, doing nothing, and doesn't even have a name!)

And lastly, here's the anonymous function: 

```js
(num1, num2) => num1 + num2
```

Which is the same as:
```js
(num1, num2) => { return num1 + num2; }
```

## Putting it all together

So what you're doing with this line is retrieving this function:

```js
(num1, num2) => num1 + num2
```
 from `MATH_FUNCS`

and then RUNNING that function by supplying the `()` to it. 
But also, while you were running the function, you supplied the parameters to it that it needs to run _properly_. 

```js
MATH_FUNCS["add"](3, 4)
```

Reply:
It WORKS! Removed the returns, no errors. Thanks!

Reply:
Added this and appears to be working!